src/EnergyPlus/BaseboardRadiator.hh

```diff
@@ -155,19 +134,12 @@ namespace BaseboardRadiator {
 struct BaseboardRadiatorData : BaseGlobalStruct
 {
 
-    int NumBaseboards = 0;
     bool getInputFlag = true;
-    bool ZoneEquipmentListChecked = false;
-    Array1D<BaseboardRadiator::BaseboardParams> Baseboard;
-    Array1D<BaseboardRadiator::BaseboardParamsNumericFieldData> BaseboardParamsNumericFields;
+    EPVector<BaseboardRadiator::BaseboardParams> baseboards;
 
     void clear_state() override
     {
-        this->NumBaseboards = 0;
-        this->getInputFlag = true;
-        this->ZoneEquipmentListChecked = false;
-        this->Baseboard.deallocate();
-        this->BaseboardParamsNumericFields.deallocate();
+        *this = BaseboardRadiatorData();
```

Comment:
@Myoldmopar Explain to me how this works? Please.

Reply:
Good question.  The way we are doing data classes right now is super risky from an "accidentally introducing bugs" perspective.  Right now we declare a bunch of variables, and either initialize them at the declaration, or in the constructor.  Then inside the clear_state function, we are just manually going back over them and reinitializing them all to the original values.  This is just begging for someone to change one location but forget the other and lead to strange issues.

What we really want is to simply get a fresh instance of the data class.  We don't need to manually go over each variable to reassign a value, just give me a fresh instance that will have all the variables set to the initial data automatically.  That's what this line does.  Behind the scenes, the assignment operator is actually calling the constructor and destructor of the class [as seen here](https://godbolt.org/z/najqc551o), but this is only happening in between unit tests or in between multiple runs with the API, **nothing at all regarding the runtime of a single simulation**.

There is actually a side benefit of this method, which is sniffing out things in state.  For example, if the data class has a const member, then the default assignment operator gets implicitly deleted and it won't compile.  This is great for us, because const data does not need to live in state.  It is a big red flag that that member variable does not need to be in there.  This also goes for those few things that we manually manage the memory of, such as the python plugins.  Again, this pattern will just raise a red flag to focus us to certain patterns that need to be either cleaned up or need special attention.

Now, all that said, this type of assignment does not really have a place within the simulation code.  At least not that I can think of.  We wouldn't want to be calling constructors and destructors lots of times in the code.  So I wouldn't worry about this much.  Also -- it's possible that once we get these all cleaned up, it will be easy to just eliminate the clear_state methods entirely and get new instances at a higher level.  Even less code to maintain...